data/mods/mixandmega/random-teams.ts

```diff
@@ -0,0 +1,184 @@
+				species = dex.species.get(species.changesFrom);
+				forme = species.name;
+			}
+
+			// Random legal ability
+			const abilities = Object.values(species.abilities).filter(a => this.dex.abilities.get(a).gen <= this.gen);
+			const ability: string = this.gen <= 2 ? 'No Ability' : this.sample(abilities);
+
+			// Four random unique moves from the movepool
+			let pool = ['struggle'];
+			if (forme === 'Smeargle') {
+				pool = this.dex.moves
+					.all()
+					.filter(move => !(move.isNonstandard || move.isZ || move.isMax || move.realMove))
+					.map(m => m.id);
+			} else {
+				const formes = ['gastrodoneast', 'pumpkaboosuper', 'zygarde10'];
+				let learnset = this.dex.species.getLearnset(species.id);
+				let learnsetSpecies = species;
+				if (formes.includes(species.id) || !learnset) {
+					learnsetSpecies = this.dex.species.get(species.baseSpecies);
+					learnset = this.dex.species.getLearnset(learnsetSpecies.id);
+				}
+				if (learnset) {
+					pool = Object.keys(learnset).filter(
+						moveid => learnset![moveid].find(learned => learned.startsWith(String(this.gen)))
+					);
+				}
+				if (learnset && learnsetSpecies === species && species.changesFrom) {
+					learnset = this.dex.species.getLearnset(toID(species.changesFrom));
+					for (const moveid in learnset) {
+						if (!pool.includes(moveid) && learnset[moveid].some(source => source.startsWith(String(this.gen)))) {
+							pool.push(moveid);
+						}
+					}
+				}
+				const evoRegion = learnsetSpecies.evoRegion && learnsetSpecies.gen !== this.gen;
+				while (learnsetSpecies.prevo) {
+					learnsetSpecies = this.dex.species.get(learnsetSpecies.prevo);
+					for (const moveid in learnset) {
+						if (!pool.includes(moveid) &&
+							learnset[moveid].some(source => source.startsWith(String(this.gen)) && !evoRegion)) {
+							pool.push(moveid);
+						}
+					}
+				}
+			}
+
+			const moves = this.multipleSamplesNoReplace(pool, this.maxMoveCount);
+
+			// Random EVs
+			const evs: StatsTable = {hp: 0, atk: 0, def: 0, spa: 0, spd: 0, spe: 0};
+			const s: StatID[] = ["hp", "atk", "def", "spa", "spd", "spe"];
+			let evpool = 510;
+			do {
+				const x = this.sample(s);
+				const y = this.random(Math.min(256 - evs[x], evpool + 1));
+				evs[x] += y;
+				evpool -= y;
+			} while (evpool > 0);
+
+			// Random IVs
+			const ivs = {
+				hp: this.random(32),
+				atk: this.random(32),
+				def: this.random(32),
+				spa: this.random(32),
+				spd: this.random(32),
+				spe: this.random(32),
+			};
+
+			// Random nature
+			const nature = this.sample(natures).name;
+
+			// Level balance--calculate directly from stats rather than using some silly lookup table
+			const mbstmin = 1307; // Sunkern has the lowest modified base stat total, and that total is 807
+
+			let stats = species.baseStats;
+			// If Wishiwashi, use the school-forme's much higher stats
+			if (species.baseSpecies === 'Wishiwashi') stats = Dex.species.get('wishiwashischool').baseStats;
```

Comment:
```suggestion
			let stats = species.baseStats;
			// If Wishiwashi, use the school-forme's much higher stats
			if (species.baseSpecies === 'Wishiwashi') stats = Dex.species.get('wishiwashischool').baseStats;
```
Should you take into account the post-mega stats here?